lib/spack/spack/binary_distribution.py

```diff
@@ -145,6 +152,7 @@ def write_buildinfo_file(prefix, workdir, rel=False):
         prefix, spack.store.layout.root)
     buildinfo['relocate_textfiles'] = text_to_relocate
     buildinfo['relocate_binaries'] = binary_to_relocate
+    buildinfo['relocate_links'] = link_to_relocate
```

Comment:
I'm curious why this data is retained to be handled later, after the binary cache is unpacked.

The binaries/textfiles, may contain absolute links, they are currently maintained as such, so therefore this bookkeeping is necessary. Is there any reason the binary cache cannot be created with the symlinks relative-ized?

Reply:
I opted (without much reason at the time) to maintain the idiom of relativizing or not depending on whether the rpaths were being relativized.

In the context of your comment above about python link resolution, I think I now have a good reason to keep things this way. I expect there will be packages that break if their links are relativized, given python's broken link resolution.

Reply:
> I expect there will be packages that break if their links are relativized, given python's broken link resolution.

I think we can resolve this ourselves as long as we avoid calling `realpath`. I think that should be avoided anyhow given the concerns I voice elsewhere in the review.